src/extra/gizmos/context.cpp

```diff
@@ -29,25 +30,24 @@ struct HelperContextShard : public gfx::BaseConsumer {
   void warmup(SHContext *context) {
     baseConsumerWarmup(context, true);
 
-    // Reference and setup context
+    // Reference context variable
     _contextVarRef = referenceVariable(context, Context::contextVarName);
-    _contextVarRef->payload.objectValue = &_context;
-    _contextVarRef->payload.objectVendorId = SHTypeInfo(Context::Type).object.vendorId;
-    _contextVarRef->payload.objectTypeId = SHTypeInfo(Context::Type).object.typeId;
-    _contextVarRef->valueType = SHType::Object;
 
-    PARAM_WARMUP(context);
+    {
+      ObjectVarScope _scope(*_contextVarRef, &_context, Context::Type);
+      PARAM_WARMUP(context);
+    }
   }
 
   void cleanup() {
-    baseConsumerCleanup();
+    {
+      ObjectVarScope _scope(*_contextVarRef, &_context, Context::Type);
+      PARAM_CLEANUP();
+    }
 
-    PARAM_CLEANUP();
+    baseConsumerCleanup();
 
     if (_contextVarRef) {
-      if (_contextVarRef->refcount > 1) {
```

Comment:
this is not ideal, we already have leaks which I don't know you fixed yet or not. Generally these red flags are the culprit...

Reply:
Can you instead keep this and fixup refcount when you really know what is going on behind the scene? We do that in (Ref) for example

Reply:
To be clear, those checks exits to check external behaviors not your own code which you know. but they are there to check they interact well with all the rest.

Reply:
@sinkingsugar I'm not sure what you mean with fixing up refcount.
In case there's 2 of these contexts, they'll both reference the same variable. I can't really know when the variable is no longer used by either.

Reply:
We already have leaks for months and these are red flags as well. I don't have the time to check sadly in depth. So I trust you do the right thing.